### ISSUE TITLE
dynamo/torchxla integration handle dropout incorrectly for training

### ISSUE THREAD

**Issue by shunting314** [pytorch/pytorch#90097]
### 🐛 Describe the bug

For this simple model
```
class DropoutModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear = nn.Linear(10, 10)
        self.dropout = nn.Dropout(p=0.1)

    def forward(self, x):
        x = self.linear(x)
        x = self.dropout(x)
        return x

    def get_example_inputs(self):
        return (torch.randn(10),)
```

Running this thru the aot_torchxla_trace_once dynamo backend (need patch https://github.com/pytorch/pytorch/pull/88449), we would see that test and baseline dropout different items. This may be an issue with random seed.

Command:
```
USE_FAKE_TENSOR=0 GPU_NUM_DEVICES=1 python benchmarks/dynamo/torchbench.py --randomize-input --performance --trace-on-xla --training --backend=aot_torchxla_trace_once -n1 --only path:/pytorch/myscripts/model_collection.py,class:DropoutModel
```

### Versions

..

cc @ezyang @gchanan @zou3519 @bdhirsh @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

**Comment by ezyang** [comment]
In inductor, this is expected because we didn't exactly replicate RNG behavior (though, IMO, we really should). Not sure about torchxla though.

**Comment by shunting314** [comment]
@ezyang  we are trying to force sync the xla random seed for baseline and test (by calling `xla_model.set_rng_state`), that works for the toy models in the issue summary, but still fail model like `mobilenet_v2`. Digging more into it.

**Comment by shunting314** [comment]
here is my fix: https://github.com/pytorch/xla/pull/4293
